config/routes.rb

```diff
@@ -72,6 +72,9 @@
   put   'article/article_category', :to => 'article#article_category'
   patch 'category/update', :to => 'category#update'
   patch 'user/update', :to => 'user#update'
+  get   ':collection_id/article/upload', :to => 'article#upload_form', :as => 'article_upload_form'
+  post  'article/upload', :to => 'article#subject_upload', :as => 'article_subject_upload'
+  get   '/article/upload_example', :to => 'article#upload_example', :as => 'article_upload_example'
```

Comment:
The equivalent metadata.yml doesn't seem to have a route, and just goes to static/metadata.yml.  (I don't care that much, but it seems simpler.)

Reply:
I think we're going to have to have specific routes for everything; `rails6` seems to expect them.